ResearchKit/ActiveTasks/ORKDeviceMotionReactionTimeViewController.m

```diff
@@ -80,8 +80,23 @@ - (void)viewWillDisappear:(BOOL)animated {
 - (void)start {
     [super start];
     [self startStimulusTimer];
+
 }
 
+#if TARGET_IPHONE_SIMULATOR
+- (void)motionBegan:(UIEventSubtype)motion withEvent:(UIEvent *)event
+{
+    if(event.type == UIEventSubtypeMotionShake)
+    {
+        ORKDeviceMotionReactionTimeResult *reactionTimeResult = [[ORKDeviceMotionReactionTimeResult alloc] initWithIdentifier:self.step.identifier];
+        reactionTimeResult.timestamp = _stimulusTimestamp;
+        [_results addObject:reactionTimeResult];
```

Comment:
This needs an `if (_validResult)`check before adding the result to the results array, same as with `recorder:didCompleteWithResult:`.

Otherwise, triggering a shake gesture while the blue dot is not showing stores a valid result and consumes and attempt (even though the failure animation is correctly shown).